enumerable_methods.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # my_each
+  def my_each
+    return .to_enum unless block_given?
+    if self.class == Array
+      i = 0
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      keys.length.times do |item|
+        key = keys[item]
+        value = self[key]
+        yield(key, value)
+      end
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    if self.class == Array
+      i = 0
+      while i < length
+        yield(self[i], i)
+        i += 1
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      keys.length.times do |item|
+        key = keys[item]
+        value = self[key]
+        key_value = [key, value]
+        yield(key_value, item)
+      end
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_arr = []
+    i = 0
+    while i < size
+      new_arr << self[i] if yield(self[i])
+      i += 1
+    end
+    new_arr
+  end
+
+  # my_all
+  def my_all?
+    return my_all? { |obj| obj } unless block_given?
+
+    if self.class == Array
+      length.times do |i|
+        return false unless yield(self[i])
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      length.times do |i|
+        return false unless yield(keys[i], self[keys[i]])
+      end
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* When a class is passed as an argument returns true if all of the collection is a member of such class.
```
Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true if all of the collection matches the Regex.
```
Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
```
Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```